### PR TITLE
Adding monitoring reader role to AMW in Arm template

### PR DESCRIPTION
**Pull request by Sohamdg081992**
This pr adds the monitor reader role to Azure Monitor Workspace in the Arm template. 

Testing: Tested deployment in personal AMW https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/ce4d1293-71c0-4c72-bc55-133553ee9e50/resourceGroups/sohamTestKV/providers/microsoft.monitor/accounts/sohamtestpolicy/users

### CODE REVIEW

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json [hunk 1]
```diff
@@ -345,6 +352,16 @@
         ]
       }
     },
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
+      "apiVersion": "2022-04-01",
+      "name": "[parameters('roleNameGuid')]",
+      "scope": "[parameters('azureMonitorWorkspaceResourceId')]",
+      "properties": {
+          "roleDefinitionId": "[concat('/subscriptions/', variables('clusterSubscriptionId'), '/providers/Microsoft.Authorization/roleDefinitions/', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')]",
```
Comment (bragi92) [inline AddonArmTemplate/FullAzureMonitorMetricsProfile.json:361]: Please update this to the GUID for the Monitoring Data Reader Role.

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json [hunk 2]
```diff
@@ -345,6 +352,16 @@
...
+          "principalId": "[reference(resourceId('Microsoft.Dashboard/grafana', split(parameters('grafanaResourceId'),'/')[8]), '2022-08-01', 'Full').identity.principalId]"
```
Comment (bragi92) [inline AddonArmTemplate/FullAzureMonitorMetricsProfile.json:362]: Can you also create a PR to update the ARM template documenation to remove the extra role assignment step and mention the permission level required for customers to enable this?

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json [hunk 3]
```diff
@@ -345,6 +352,16 @@
         ]
       }
     },
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
```
Comment (bragi92) [inline AddonArmTemplate/FullAzureMonitorMetricsProfile.json:356]: Can you update the bicep template too in this PR?

**Review by bragi92 — CHANGES_REQUESTED** [review]

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json (hunk 1 above)
Comment (Sohamdg081992) [reply AddonArmTemplate/FullAzureMonitorMetricsProfile.json:361]: Sure. Thanks!

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json (hunk 2 above)
Comment (Sohamdg081992) [reply AddonArmTemplate/FullAzureMonitorMetricsProfile.json:362]: Sure! I am doing this in the next commit in the prometheus-collector repo. I will create a separate pr for updating the public docs in azure-pr-docs repo.

File: AddonArmTemplate/FullAzureMonitorMetricsProfile.json (hunk 3 above)
Comment (Sohamdg081992) [reply AddonArmTemplate/FullAzureMonitorMetricsProfile.json:356]: Yes, I am updating the bicep in this pr.

File: AddonBicepTemplate/FullAzureMonitorMetricsProfile.bicep [hunk 4]
```diff
@@ -270,6 +280,14 @@ resource kubernetesRecordingRuleGroup 'Microsoft.AlertsManagement/prometheusRule
   }
 }
 
+resource roleNameGuid_resource 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: roleNameGuid
+  properties: {
```
Comment (Sohamdg081992) [inline AddonBicepTemplate/FullAzureMonitorMetricsProfile.bicep:285]: Currently in bicep, there is no way to explicitly "scope" the role assignment on a "resource id"(like in Arm template). Bicep expects a value of type "resource | tenant" and currently there is no resource spec for Azure Monitor Workspace. So, as a workaround, the default scoping is on the resource group and thus the monitor reader role is applied on the same Azure monitor workspace(by inheritence) which is the expected behavior. I have tested it in my workspace [here](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/ce4d1293-71c0-4c72-bc55-133553ee9e50/resourceGroups/sohamTestKV/providers/microsoft.monitor/accounts/sohambiceptemplate/users).

Comment (bragi92) [reply AddonBicepTemplate/FullAzureMonitorMetricsProfile.bicep:285]: Sounds good, we can mention this as a limiation in the public preview doc update that you make.

**Review by bragi92 — APPROVED** [review]

File: AddonBicepTemplate/FullAzureMonitorMetricsProfile.bicep (hunk 4 above)
Comment (Sohamdg081992) [reply AddonBicepTemplate/FullAzureMonitorMetricsProfile.bicep:285]: Yes. I will mention this. Thanks!

**Pull request merged**